lib/galaxy/datatypes/isa.py

```diff
@@ -0,0 +1,395 @@
+        else:
+            dataset.name = 'ISA DATASET'
+
+    # Display data {{{2
+    ################################################################
+
+    def display_data(self, trans, dataset, preview=False, filename=None, to_ext=None, offset=None, ck_size=None, **kwd):
+        """Downloads the ISA dataset if `preview` is `False`;
+           if `preview` is `True`, it returns a preview of the ISA dataset as a HTML page.
+           The preview is triggered when user clicks on the eye icon of the composite dataset."""
+
+        self._set_dataset_name(dataset)
+        # if it is not required a preview use the default behaviour of `display_data`
+        if not preview:
+            return super(_Isa, self).display_data(trans, dataset, preview, filename, to_ext, **kwd)
+
+        # prepare the preview of the ISA dataset
+        investigation = self._get_investigation(dataset)
+        if investigation is None:
+            html = """<html><header><title>Error while reading ISA archive.</title></header>
+                   <body>
+                        <h1>An error occured while reading content of ISA archive.</h1>
+                        <p>If you have tried to load your archive with the uploader by selecting isa-tab as composite data type, then try to load it again with isa-json instead. Conversely, if you have tried to load your archive with the uploader by selecting isa-json as composite data type, then try isa-tab instead.</p>
+                        <p>You may also try to look into your zip file in order to find out if this is a proper ISA archive. If you see a file i_Investigation.txt inside, then it is an ISA-Tab archive. If you see a file with extension .json inside, then it is an ISA-JSON archive. If you see nothing like that, then either your ISA archive is corrupted, or it is not an ISA archive.</p>
+                   </body></html>"""
+        else:
+            html = '<html><body>'
+            html += '<h1>{0} {1}</h1>'.format(investigation.title, investigation.identifier)
+
+            # Loop on all studies
+            for study in investigation.studies:
+                html += '<h2>Study %s</h2>' % study.identifier
+                html += '<h3>%s</h3>' % study.title
+                html += '<p>%s</p>' % study.description
+                html += '<p>Submitted the %s</p>' % study.submission_date
+                html += '<p>Released on %s</p>' % study.public_release_date
+
+                # Loop on all assays of this study
+                for assay in study.assays:
+                    html += '<h3>Assay %s</h3>' % assay.filename
+                    html += '<p>Measurement type: %s</p>' % assay.measurement_type.term  # OntologyAnnotation
+                    html += '<p>Technology type: %s</p>' % assay.technology_type.term    # OntologyAnnotation
+                    html += '<p>Technology platform: %s</p>' % assay.technology_platform
+                    if assay.data_files is not None:
```

Comment:
Note that this bit doesn't actually output any of the assay data files as it stands, because you're only using the investigation file parser from `isa-rwval`. So in all cases this data files list will be empty

Reply:
Yes, I've noted that it's always empty now. That's unfortunate. Any workaround?

Reply:
I think this has become like this because the original request for `isa-rwval` was to support Python 2.7 (since full `isatools` is Python 3), but then it turns out that for Galaxy we don't want to add `pandas` to the requirements. The study and assay table parsing relies on `pandas` currently.

I can probably add to `isa-rwval` some features to extract the data files (and possibly a little bit more other useful metadata). Once I push a new version, the pinned-requirements.txt may need to be updated.

Reply:
I can do it today.

Reply:
Great job @djcomlab , I've just tested the modifications on my computer. It works well.